apps/www/registry/tui/ui/textarea.tsx

```diff
@@ -1,21 +1,441 @@
-  extends React.TextareaHTMLAttributes<HTMLTextAreaElement> {}
+  extends React.TextareaHTMLAttributes<HTMLTextAreaElement>,
+  VariantProps<typeof textAreaVariants> {
+  asChild?: boolean;
+  label?: string | undefined;
+  image?: string | undefined;
+  showIconForTextArea?: boolean | undefined;
+  writeButton?: string | undefined;
+  previewButton?: string | undefined;
+  submitButton?: string | undefined;
+  showMoodButton?: 'default' | "resultButton";
+  textAreaButtonvariant?: "outline" | "gray";
+  placeholder?: string | undefined;
+  variant?: 'outline';
+  textareavariant?: 'default' | 'borderForInput';
+  icons?: IconType[] | undefined;
+  icon?: IconType | undefined;
+  titlePlaceholder?: string | undefined;
+  hasDivider?: boolean | undefined;
+  assignList?: AssignListItem[] | undefined;
+  assignButtonName?: string | undefined;
+  labelButtonName?: string | undefined;
+  labelList?: LabelListItem[] | undefined;
+  dueDateButtonName?: string | undefined;
+  dueDateList?: DueDateListItem[] | undefined;
+  attachIcon?: IconType | undefined;
+  attachFileName?: string | undefined;
+  iconButton?: IconButton[] | undefined;
+  previewContent?: string | undefined;
+  createButton?: string | undefined;
+  iconStyle?:string | undefined;
+  submitButtonName?:string | undefined;
+  color?: "black" | "white" | "slate" | "gray" | "zinc" | "neutral" | "stone" |
+  "red" | "orange" | "amber" | "yellow" | "lime" | "green" | "emerald" | "teal" | "cyan"
+  | "sky" | "blue" | "indigo" | "violet" | "purple" | "fuchsia" | "pink" | "rose";
+}
+
 
 const Textarea = React.forwardRef<HTMLTextAreaElement, TextareaProps>(
-  ({ className, ...props }, ref) => {
+  ({ className, showIconForTextArea,color,submitButtonName, iconButton,iconStyle, createButton, attachFileName, previewContent, labelList, attachIcon, dueDateList, dueDateButtonName, assignList, labelButtonName, assignButtonName, hasDivider, titlePlaceholder, icon, variant, icons, textareavariant, placeholder, showMoodButton, textAreaButtonvariant, previewButton, submitButton, writeButton, label, ...props }, ref) => {
+    const [showComment, setShowComment] = useState(false);
+    const [showWrite, setShowWrite] = useState(false);
+    const [showIcon, setShowIcon] = useState(false);
+    const [selectedIcon, setSelectedIcon] = useState<IconList | null>(null);
+    const [isListVisible, setIsListVisible] = useState(false);
+    const [isLabelListVisible, setIsLabelListVisible] = useState(false);
+    const [isDueDateListVisible, setIsDueDateListVisible] = useState(false);
+    const [selectedOption, setSelectedOption] = useState<AssignListItem | null>(null);
+    const [selectedLabelOption, setSelectedLabelOption] = useState<LabelListItem | null>(null);
+    const [selectedDueDateOption, setSelectedDueDateOption] = useState<DueDateListItem | null>(null);
+
+    const buttonColor = (color?: string) => {
+      return ` bg-${color}-600 text-white`
+    }
+
+    const handleCommentButtonClick = () => {
```

Comment:
wrap these in useCallbacks

Reply:
for example:  
   const handleWriteButtonClick = useCallback(() => {
      memoizedSetShowWrite(true);
      memoizedSetShowComment(false);
    }, []);
    
    It memoize the functions which returns the same result each time component renders.
    Make sure to momoize such functions or constants